internal/translation/customroles/custom_roles.go

```diff
@@ -0,0 +1,63 @@
+package customroles
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+)
+
+type CustomRoleService interface {
+	Get(ctx context.Context, projectID string, roleName string) (*CustomRole, error)
+	List(ctx context.Context, projectID string) ([]CustomRole, error)
+	Create(ctx context.Context, projectID string, role CustomRole) error
+	Update(ctx context.Context, projectID string, roleName string, role CustomRole) error
+	Delete(ctx context.Context, projectID string, roleName string) error
+}
+
+type CustomRoles struct {
+	roleAPI admin.CustomDatabaseRolesApi
+}
+
+func NewCustomRoles(api admin.CustomDatabaseRolesApi) *CustomRoles {
+	return &CustomRoles{roleAPI: api}
+}
+
+func (s *CustomRoles) Get(ctx context.Context, projectID string, roleName string) (*CustomRole, error) {
+	customRole, _, err := s.roleAPI.GetCustomDatabaseRole(ctx, projectID, roleName).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get custom roles from Atlas: %w", err)
+	}
+
+	return fromAtlas(customRole), err
+}
+
+func (s *CustomRoles) List(ctx context.Context, projectID string) ([]CustomRole, error) {
+	atlasRoles, _, err := s.roleAPI.ListCustomDatabaseRoles(ctx, projectID).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list custom roles from Atlas: %w", err)
+	}
+
+	customRoles := make([]CustomRole, len(atlasRoles))
+
+	for i, r := range atlasRoles {
+		customRoles[i] = *fromAtlas(&r)
```

Comment:
I would avoid the struct copy operation and also avoid the linter issue at the same time probably.

```suggestion
	customRoles := make([]*CustomRole, len(atlasRoles))

	for i, r := range atlasRoles {
		customRoles[i] = fromAtlas(&r)
```